BC Transition Model Term2 update

sametcaka opened the pull request:
## Proposed Changes
Updated the Term2 of the BC Transition model, which eliminates some deficiencies in the previous version.The update is based on:

1- Cakmakcioglu, S. C., Bas, O., Mura, R., and Kaynak, U., "A Revised One-Equation Transitional Model for External Aerodynamics," AIAA Paper 2020-2706, June 2020, https://doi.org/10.2514/6.2020-2706.
2- Mura, R. and Cakmakcioglu, S. C., "A Revised One-Equation Transitional Model for External Aerodynamics - Part I: Theory, Validation and Base Cases," AIAA Paper 2020-2714, June 2020, https://doi.org/10.2514/6.2020-2714.

which is also presented here: https://turbmodels.larc.nasa.gov/sa-bc_1eqn.html


## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*



## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [ X ] I am submitting my contribution to the develop branch.
- [ X ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [ X ] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

pcarruscag commented:
Thank you for letting us know about the update.
We will discuss whether to update or to provide both versions as BC and BCM, as described in the turbulence models page (to avoid any doubt).
If you would like to join us, the meetings take place Wednesdays at 4pm CET https://meet.jit.si/SU2_DevMeeting
By the way, are you aware of any other issues with the implementation, or have you compared the results with other references? (I suppose you are the best person to ask).

sametcaka commented:
Hello Pedro,
I am one of the developers of the BC model itself, and I have implemented it to the SU2 back in 2016 which was first available to the users on v5 "Raven" . We updated the model based on the suggestions of some experts in the field, and created the BCM version several months ago. I am using the updated version for months on my pc without any issues.
By the way, for the regression tests, I will need help as I do not have a linux machine to work with; can not compile the code properly...

TobiKattmann commented:
> as described in the turbulence models page

@pcarruscag I am unaware of this page  (not the [website](https://su2code.github.io/docs_v7/Theory/#turbulence-modeling), not a [su2code team](https://github.com/orgs/su2code/teams) :shrug: ). Can you tell me where to find it

> By the way, for the regression tests, I will need help as I do not have a linux machine to work with; can not compile the code properly...

@sametcaka just the `Schubauer Klebanoff` Testcase is failing for serial and parallel regression and [tutorials](https://su2code.github.io/tutorials/Transitional_Flat_Plate/) which you can lookup via the checks-tab above. It would be a matter of updating the regression test values for this very case (so no big deal) if the results for the updated code are what you expect.

pcarruscag commented:
The NASA one ->https://turbmodels.larc.nasa.gov/sa-bc_1eqn.html

WallyMaier commented on SU2_CFD/src/numerics/turbulent/turb_sources.cpp:
```diff
@@ -153,19 +153,19 @@ CNumerics::ResidualType<> CSourcePieceWise_TurbSA::ComputeResidual(const CConfig
       re_v   = ((Density_i*pow(dist_i,2.))/(Laminar_Viscosity_i))*Omega;
       re_theta = re_v/2.193;
       re_theta_t = (803.73 * pow((tu + 0.6067),-1.027)); //MENTER correlation
       //re_theta_t = 163.0 + exp(6.91-tu); //ABU-GHANNAM & SHAW correlation
```
Would it make sense to have an option for the different corrections?

sametcaka replied:
I do not think it would be necessary.

pcarruscag replied:
Then the commented code should be deleted, comments have a tendency for obsolescence.
Also around line 5532 of CConfig.cpp there should me a message stating the model being used has the revised coefficients.

pcarruscag commented:
To keep the authorship and traceability of the changes I will merge this with the failing testcases and update them in a separate PR (since we have to do other transition-related fixes).
